### PR TITLE
Correct the order of Flags for Exp.Share Setup

### PR DESCRIPTION
**Pull request by wisp92**
Corrected the SetExpAllFlags method in order to set the correct flags for Exp.Share (Reversed the direction of the rotation)

I also added an extra check to determine if any participants have not fainted. I decided to share all the XP in this case for two reasons. First, I don't see how some XP cannot be shared, and second, it was easier to implement.

Finally, I commented out one seemingly not necessary check in experience.asm. I tested it and seems OK, but maybe there are implications that I am not aware of.

EDIT: I merged the Text Fix for Fixed-Damage Moves to Lite.

### CODE REVIEW

File: engine/battle/experience.asm [hunk 1]
```diff
@@ -19,8 +19,8 @@ GainExperience:
 .partyMonLoop ; loop over each mon and add gained exp
 	inc hl
 	ld a, [hli]
-	or [hl] ; is mon's HP 0?
-	jp z, .nextMon ; if so, go to next mon
+	; or [hl] ; is mon's HP 0? wispnote - We already handle this check separately, also makes debugging more difficult.
```
Comment (jojobear13) [inline engine/battle/experience.asm:22]: These lines are likely needed to handle unpredictable one-off cases

File: engine/battle/stats_functions.asm [hunk 2]
```diff
@@ -444,12 +444,10 @@ SetExpAllFlags:
...
 	ld a, [hli]
 	or [hl] ; is mon's HP 0?
-	jp z, .setzeroexpflag
+	jp z, .setnextexpflag
 	scf
-	rl b
-	jp .nextmonforexpall
-.setzeroexpflag
-	sla b
+.setnextexpflag
```
Comment (jojobear13) [inline engine/battle/stats_functions.asm:449]: There we go. Now in-line comments are working. Anyway, assumes the party is always 6.

Comment (wisp92) [reply engine/battle/stats_functions.asm:449]: Thanks, for the correction I fixed the issue in the latest commit.

File: engine/battle/stats_functions.asm [hunk 3]
```diff
@@ -460,6 +459,16 @@ SetExpAllFlags:
...
+	ld a, [wPartyCount]
+	ld c, a
+	ld a, 6
+	sub c
+.loopBitRotation
+	jp z, .finishVariableBitRotation
+	rr b
+	dec a
+	jr .loopBitRotation
+.finishVariableBitRotation
```
Comment (jojobear13) [inline engine/battle/stats_functions.asm:471]: This wont work. Imagine a party of 6 and this is your first time in the loop. A becomes 6, C becomes 6, and your result is zero so you do two rr commands. The carry value starts in the carry bit. The first rr moves it to bit 7. The second rr moves it to bit 6. wPartyCount is also not decrementing so on the next wMon loop you are always doing 6 - 6. The second mon has its flag value now in bit 6 and the first mon has its flag value now in bit 4.

Comment (wisp92) [reply engine/battle/stats_functions.asm:471]: Did you account for the rr b instructions before the loop?

**Comment by wisp92** [comment]
I updated the request in accordance to our discussion. Please review.

**Comment by jojobear13** [comment]
Closing this. I figured out how to cherry-pick commits and I've been picking through the commits.

I really appreciate your work. You really know your stuff. But I've found that github pulls can become unwieldy very quickly. In the future, it would help me a lot to limit pulls to a single concrete commit. A lot of smaller pulls would be much easier to manage.

**Comment by wisp92** [comment]
Will do. Also, thanks. It's good to hear that since this is the first time I use Github for collaboration. This is also the first time I study assembly, and it's been quite fun to read the code so far. I never tried something lower than C++ before.

**Pull request closed without merging**